### PR TITLE
Adding a CHANGELOG (replacing old PR)

### PR DESCRIPTION
**Pull request by Kully**
I have done some substantial work to the CHANGELOG and would like some feedback before going any further.

**What has been done?**
- remove most of the bullet points under `Other Changes` and got rid of that section (except for the last section in the CHANGELOG). Most of these changes are not important for users to know about in a CHANGELOG
- several times combined `- Supporting foo` and `-Added bar` lines together if of the same type (i.e. Unicode Character, a \method, etc) and deleted the associated the issues/PRs. I like this decision as it conserves space, makes it easier to read, and the issues can be found in the Release Section already
- put methods into code environment to look pretty eg. sqrt -> `\sqrt`

**Questions:**
1. Should we keep some of the links to the Issues/PRs?

I think Issue links are more relevant for users than the PR ones. I hyperlinked some of them but held off on doing them all.
 
2. Keep the `@flow` comments?

I've grouped `@flow` additions/changes in here. I don't know how important they are.

3. Worth making Present or Past Tense consistency in CHANGELOG?
The headers suggested from [keepachangelog.com](https://keepachangelog.com/en/1.0.0/) are in the past tense, but some of the lines in our CHANGELOG are in the present tense: eg. `-Add !@#$ ...`

Overall I think this is useful for KaTeX users rather than siffting through the denser Release Section.

Any comments or suggestions? 
cc. @kevinbarabash @ylemkimon

### CODE REVIEW

**Comment by Kully** [comment]
Closed PR: https://github.com/Khan/KaTeX/pull/1237

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+- Added support for \not (#140)
+
+### Changed
+- Support CJK full-width punctuation + Unicode dots. See [#814](https://github.com/Khan/KaTeX/pull/814)
+- Support for ```' \` ^ ~ = \u . " \r \H \v``` text-mode accents. See [#802](https://github.com/Khan/KaTeX/pull/802)
+- Modernize font creation. See [#624](https://github.com/Khan/KaTeX/pull/624)
+
+### Fixed
+- Use inline SVG for stretchy elements (#807)
+Improve `\sqrt` #810
```
Comment (kevinbarabash) [inline CHANGELOG.md]: `- Improve `\sqrt` #810`

**Comment by kevinbarabash** [comment]
> Should we keep some of the links to the Issues/PRs?
> I think Issue links are more relevant for users than the PR ones. I hyperlinked some of them but held off on doing them all.

Hmm... that's annoying that the links have to be inserted in this file.  We could leave them out for now and then maybe write a script to automate the process later.

> Keep the @flow comments?
> I've grouped @flow additions/changes in here. I don't know how important they are.

I thought that was a nice touch.

> Worth making Present or Past Tense consistency in CHANGELOG?
> The headers suggested from keepachangelog.com are in the past tense, but some of the lines in our CHANGELOG are in the present tense: eg. -Add !@#$ ...

That's a good point.  Yeah past tense is better.  I'll change things to past tense for future releases.

**Comment by Kully** [comment]
> We could leave them out for now and then maybe write a script to automate the process later.

Some of the links are for PRs, some are for Issues.

I don't mind putting in the links myself for now. If we chose to write a script to scrape the release section I don't think this would be an effective solution because:

1. We are only including relevant notes in the CHANGELOG so it would need to know which comments _are_ getting included. It's more of a human job.
2. If we are producing a changelog then checking the release notes for the changelog changes that were taken from there, it may be difficult if sentences are in past tense, restructuring of the lines, etc.

Did you have another approach in mind?

I'm gonna try to get this PR ready by the end of the weekend fyi

**Comment by kevinbarabash** [comment]
@Kully looks good, but it appears you've made a change to `submodules/katex-fonts` by accident which is causing the tests to fail.  Can you revert that change?

**Comment by ylemkimon** [comment]
You should update submodules after checking out, i.e., `git submodule update`.

**Comment by Kully** [comment]
> You should update submodules after checking out, i.e., git submodule update.

I ran this and the command did not produce any modifications to any files.

**Comment by kevinbarabash** [comment]
@Kully I think the easiest thing to do would be to create new PR and just copy the file over.  The only thing that should be in your PR is the CHANGELOG.md file right?

**Comment by ylemkimon** [comment]
Closing in favor of #1322.

**Pull request closed without merging**